pkg/kubelet/allocation/allocation_manager_test.go

```diff
@@ -1020,6 +1020,271 @@ func TestHandlePodResourcesResizeWithSwap(t *testing.T) {
+		},
+		{
+			name:                            "IPPR Disabled - New pod admitted but allocated resources not updated",
+			ipprFeatureGate:                 false,
+			initialAllocatedResourcesState:  map[types.UID]v1.ResourceList{},
+			currentActivePods:               []*v1.Pod{},
+			podToAdd:                        pod1Small,
+			admitFunc:                       nil,
+			expectAdmit:                     true,
+			expectedAllocatedResourcesState: map[types.UID]v1.ResourceList{},
+		},
+		{
+			name:                           "IPPR Enabled - New pod not admititted due to insufficient resources",
+			ipprFeatureGate:                true,
+			initialAllocatedResourcesState: map[types.UID]v1.ResourceList{pod1UID: cpu1Mem1G},
+			currentActivePods:              []*v1.Pod{pod1Small},
+			podToAdd:                       pod2Large,
+			admitFunc: func(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+				cpuRequest := attrs.Pod.Spec.Containers[0].Resources.Requests.Cpu().Value()
+				if cpuRequest > 1 {
+					return lifecycle.PodAdmitResult{
+						Admit:   false,
+						Reason:  "OutOfcpu",
+						Message: fmt.Sprintf("not enough CPUs available for pod %s/%s, requested: %d, available:1", attrs.Pod.Namespace, attrs.Pod.Name, cpuRequest),
+					}
+				}
+				return lifecycle.PodAdmitResult{Admit: true}
+			},
+			expectAdmit:             false,
+			admissionFailureReason:  "OutOfcpu",
+			admissionFailureMessage: "not enough CPUs available for pod ns2/pod2, requested: 2, available:1",
+			// allocated resources not modified
+			expectedAllocatedResourcesState: map[types.UID]v1.ResourceList{pod1UID: cpu1Mem1G},
+		},
+		{
+			name:                           "IPPR Disabled - New pod not admitted due to insufficient resources",
+			ipprFeatureGate:                false,
+			initialAllocatedResourcesState: map[types.UID]v1.ResourceList{pod1UID: cpu1Mem1G},
+			currentActivePods:              []*v1.Pod{pod1Small},
+			podToAdd:                       pod2Large,
+			admitFunc: func(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+				cpuRequest := attrs.Pod.Spec.Containers[0].Resources.Requests.Cpu().Value()
+				if cpuRequest > 1 {
+					return lifecycle.PodAdmitResult{
+						Admit:   false,
+						Reason:  "OutOfcpu",
+						Message: fmt.Sprintf("not enough CPUs available for pod %s/%s, requested: %d, available:1", attrs.Pod.Namespace, attrs.Pod.Name, cpuRequest),
+					}
+				}
+				return lifecycle.PodAdmitResult{Admit: true}
+			},
+			expectAdmit:             false,
+			admissionFailureReason:  "OutOfcpu",
+			admissionFailureMessage: "not enough CPUs available for pod ns2/pod2, requested: 2, available:1",
+			// allocated resources not modified
+			expectedAllocatedResourcesState: map[types.UID]v1.ResourceList{pod1UID: cpu1Mem1G},
+		},
+		{
+			name:                            "IPPR Enabled - pod resize request feasible. Request same as existing allocation",
```

Comment:
if request == allocation, that means there is no resize request that is still pending allocation

```suggestion
			name:                            "IPPR Enabled - no pod resize request. Resource request same as existing allocation",
```